### PR TITLE
Vehicle update2 Part 1: Water Update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Implements `VehicleHead::update()` up to the `waterUpdate()` function.

Based on #136.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Really nice to see some work on these vehicle functions again.

I notice `Vehicle.cpp` is declaring `using namespace OpenLoco`. I think instead its contents should be _in_ `namespace OpenLoco`, or perhaps even `OpenLoco::Vehicle`?

**Comment by duncanspumpkin** [comment]
Okay think thats all I will do in this PR. I'll not be adding any more to this one.

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 1]
```diff
@@ -298,4 +367,249 @@ namespace OpenLoco::Vehicles
         vehStatus.status2Args = regs.edx;
         return vehStatus;
     }
+
+    // 0x004A8882
+    void VehicleHead::updateTrainSounds()
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:372]: Update driving sounds. Need to differentiate it from start movement sounds.

**Comment by AaronVanGeffen** [comment]
On this PR's branch, boats no longer have wakes. They are rendered correctly on current master.

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
As of this PR, boats no longer have wakes. They are rendered correctly on current master.

**Comment by AaronVanGeffen** [comment]
Okay, here's how I went about testing the waters. I let the game run on a very busy save for a two in-game months, autosaving every month. I first compared two runs of this branch to itself. The results are fairly deterministic, save for a few bytes. I then did the same thing on master. Again, the results are were deterministic. Finally, I compared the autosaves from one branch to the other. I am elated to say this is yielding positive results.

#### This branch:
```
# First month, session 1 vs session 2
$ diff <(xxd autosave_2021-02-12_17-55-51.SV5) <(xxd autosave_2021-02-12_17-56-15.SV5) | wc -l
16
# Second month, session 1 vs session 2
$ diff <(xxd autosave_2021-02-12_17-55-40.SV5) <(xxd autosave_2021-02-12_17-56-04.SV5) | wc -l
16
```

#### master:
```
# First month, session 1 vs session 2
$ diff <(xxd master/autosave_2021-02-12_19-16-11.SV5) <(xxd master/autosave_2021-02-12_19-16-33.SV5) | wc -l
12
# Second month, session 1 vs session 2
$ diff <(xxd master/autosave_2021-02-12_19-16-23.SV5) <(xxd master/autosave_2021-02-12_19-16-44.SV5) | wc -l
16
```

#### master vs this branch:
```
# First month (this branch vs master)
$ diff <(xxd duncan/autosave_2021-02-12_17-56-04.SV5) <(xxd master/autosave_2021-02-12_19-16-33.SV5) | wc -l
16
# Second month (this branch vs master)
$ diff <(xxd duncan/autosave_2021-02-12_17-56-15.SV5) <(xxd master/autosave_2021-02-12_19-16-44.SV5) | wc -l
16
```


#### For completeness, example output:
```
$ diff <(xxd duncan/autosave_2021-02-12_17-56-15.SV5) <(xxd master/autosave_2021-02-12_19-16-44.SV5)
27895c27895
< 0006cf60: 6500 cf00 bd68 76ff 0008 2306 002e 03c0  e....hv...#.....
---
> 0006cf60: 6500 c800 bd68 76ff 0008 2306 002e 03c0  e....hv...#.....
27914c27914
< 0006d090: a897 8604 020d bc00 0180 f0fa 02a8 61ff  ..............a.
---
> 0006d090: a897 8604 0207 bc00 0180 f0fa 02a8 61ff  ..............a.
27977c27977
< 0006d480: 59b9 8700 e803 0100 1412 0003 0003 0003  Y...............
---
> 0006d480: 52b9 8700 e803 0100 1412 0003 0003 0003  R...............
172909c172909
< 002a36c0: 0087 fd07 0002 6775 1913                 ......gu..
---
> 002a36c0: 0087 fd07 0002 5375 1913                 ......Su..
```

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**